azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/FunctionApp.java

```diff
@@ -64,10 +64,11 @@ public String getMasterKey() {
     @Nonnull
     public List<FunctionEntity> listFunctions(boolean... force) {
         return Optional.ofNullable(this.getFullRemote()).map(r -> r.listFunctions().stream()
-                .map(AppServiceUtils::fromFunctionAppEnvelope)
-                .filter(Objects::nonNull)
-                .collect(Collectors.toList()))
-            .orElse(Collections.emptyList());
+                        .map(envelope ->
+                                AppServiceUtils.fromFunctionAppEnvelope(envelope, (com.azure.resourcemanager.appservice.models.FunctionApp) this.getRemote()))
```

Comment:
`this.getRemote()` may send additional request to server, use `this.getId()` since only `id` is required.